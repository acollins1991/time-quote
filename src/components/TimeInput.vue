<template>
    <div>
        <input type="number" name="minutes" min="0" max="60" step="5" class="tq-time-input tq-time-input--minutes" v-bind:class="inputClasses" v-model.number="minutesInputValue">
        <input type="number" name="hours" min="0" max="25" step="1" class="tq-time-input tq-time-input--hours" v-bind:class="inputClasses" v-model.number="hoursInputValue">
        <input type="number" name="days" min="0" max="30" step="1" class="tq-time-input tq-time-input--days" v-bind:class="inputClasses" v-model.number="daysInputValue">
    </div>
</template>

<script>
export default {
    name: 'TimeInput',
    props: ['guess'],
    data() {
        return {
            inputClasses: 'shadow appearance-none border border-red py-2 px-3 text-grey-darker mb-3 leading-tight focus:outline-none focus:shadow-outline',
            minutesInputValue: 0,
            hoursInputValue: 0,
            daysInputValue: 0
        }
    },
    computed: {
        totalMinutes: function(){
            return this.minutesInputValue + ( this.hoursInputValue*60 ) + ( ( 60*24 )*this.daysInputValue )
        }
    },
    watch: {
        totalMinutes: function(value){
            this.$emit('guessUpdated', this.guess, value)
        }
    }
}
</script>

