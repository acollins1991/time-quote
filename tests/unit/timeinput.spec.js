import { shallowMount } from '@vue/test-utils'
import TimeInput from '@/components/TimeInput.vue'

describe('TimeInput.vue', () => {
  it('displays inputs for minutes, hours, days', () => {
    const wrapper = shallowMount(TimeInput)
    const minutesInput = wrapper.find('input.tq-time-input--minutes')
    expect(minutesInput.exists()).toBe(true)
    const hoursInput = wrapper.find('input.tq-time-input--hours')
    expect(hoursInput.exists()).toBe(true)
    const daysInput = wrapper.find('input.tq-time-input--days')
    expect(daysInput.exists()).toBe(true)
  })
  it('inputs convert inputs to total minutes and sets guess prop to total minutes', () => {
    const wrapper = shallowMount(TimeInput, {
      propsData: {
        guess: 0
      }
    })

    // check initial values
    expect(wrapper.vm.$data.minutesInputValue).toBe(0)
    expect(wrapper.vm.$data.hoursInputValue).toBe(0)
    expect(wrapper.vm.$data.daysInputValue).toBe(0)

    // collect inputs
    const minutesInput = wrapper.find('input.tq-time-input--minutes')
    const hoursInput = wrapper.find('input.tq-time-input--hours')
    const daysInput = wrapper.find('input.tq-time-input--days')

    // set change events and new values on inputs
    minutesInput.element.value = 5
    hoursInput.element.value = 1
    daysInput.element.value = 1
    minutesInput.trigger('input')
    expect(wrapper.vm.$data.minutesInputValue)
    hoursInput.trigger('input')
    expect(wrapper.vm.$data.hoursInputValue)
    daysInput.trigger('input')
    expect(wrapper.vm.$data.daysInputValue)
 
    // expect accurate total minutes value after input changes and set guess prop to this: 5mins = 5, 1hr = 60, 1day = 1440, total = 1505
    setTimeout(() => {
      expect(wrapper.vm.totalMinutes).toBe(1505) 
      expect(wrapper.vm.$props.guess).toBe(1505)
    }, 3000) 
  })
})
