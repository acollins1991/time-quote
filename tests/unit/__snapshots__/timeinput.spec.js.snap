// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeInput.vue renders correctly 1`] = `
<div
  class="flex flex-wrap"
>
  <div
    class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
  >
    <label
      class="block uppercase tracking-wide text-grey-darker text-sm font-bold mb-2"
      for="{$guess}-minutes-input"
    >
      Minutes
    </label>
     
    <input
      class="tq-time-input tq-time-input--minutes appearance-none block w-full bg-grey-lighter text-grey-darker border border-red rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white"
      id="{$guess}-minutes-input"
      max="60"
      min="0"
      name="minutes"
      step="5"
      type="number"
    />
  </div>
   
  <div
    class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
  >
    <label
      class="block uppercase tracking-wide text-grey-darker text-sm font-bold mb-2"
      for="{$guess}-hours-input"
    >
      Hours
    </label>
     
    <input
      class="tq-time-input tq-time-input--hours appearance-none block w-full bg-grey-lighter text-grey-darker border border-red rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white"
      id="{$guess}-hours-input"
      max="25"
      min="0"
      name="hours"
      step="1"
      type="number"
    />
  </div>
   
  <div
    class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
  >
    <label
      class="block uppercase tracking-wide text-grey-darker text-sm font-bold mb-2"
      for="{$guess}-days-input"
    >
      Days
    </label>
     
    <input
      class="tq-time-input tq-time-input--days appearance-none block w-full bg-grey-lighter text-grey-darker border border-red rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white"
      id="{$guess}-days-input"
      max="30"
      min="0"
      name="days"
      step="1"
      type="number"
    />
  </div>
</div>
`;
